addons/website_quote/models/sale_quote.py

```diff
@@ -73,15 +73,30 @@ def _onchange_product_uom(self):
         if self.product_id and self.product_uom_id:
             self.price_unit = self.product_id.uom_id._compute_price(self.product_id.lst_price, self.product_uom_id)
 
+    def _quote_line_translation(self, product_id):
+        Translation = self.env['ir.translation']
+        product_tmpl_id = self.env['product.product'].browse(product_id).product_tmpl_id.id
+        quote_translation = Translation.search([('name', '=', 'sale.quote.line,name'), ('res_id', 'in', self.ids), ('type','=', 'model')])
+        if not quote_translation:
+            product_translation = Translation.search([('name', '=', 'product.template,name'), ('res_id', '=', product_tmpl_id), ('type','=', 'model')])
+            for translation in product_translation:
+                translation.copy(default={'name': 'sale.quote.line,name', 'res_id': self.ids[0]})
```

Comment:
Please ensure it works if `self` has more than one record.
Do it the simple way, add a `for line in self` statement before the definition of `quote_translation`...